components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/dynamic_prompt.jsp

```diff
@@ -0,0 +1,136 @@
+<%--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  --%>
+
+<%@ page import="org.owasp.encoder.Encode" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.DynamicPromptUtil" %>
+<%@ page import="java.util.zip.DataFormatException" %>
+<%@ page import="java.net.URLEncoder" %>
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
+<%@include file="localize.jsp" %>
+<%@taglib prefix="e" uri="https://www.owasp.org/index.php/OWASP_Java_Encoder_Project" %>
+<jsp:directive.include file="init-url.jsp"/>
+<jsp:directive.include file="template-mapper.jsp"/>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+
+<%
+    String templateId = request.getParameter("templateId");
+    String promptId = request.getParameter("promptId");
+    String dataStr = request.getParameter("data");
+    Map data = null;
+    try {
+        data = DynamicPromptUtil.inflateJson(dataStr);
+    } catch (DataFormatException e) {
+        response.sendRedirect("retry.do");
```

Comment:
IIRC, there was a util method to get this, where the value can be overridable with config files

Reply:
util method is in identity.framework component. If we are going to use this util method, we need to add a new dependency for authentication endpoint.